src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -984,6 +985,10 @@ trait Contexts { self: Analyzer =>
 
     def isNameInScope(name: Name) = lookupSymbol(name, _ => true).isSuccess
 
+    def isPlainClassInJavaUnit(sym: Symbol): Boolean =
```

Comment:
Would it be possible/make sense to get the `unit` from `sym`, so that we can encapsulate this logic in `findMember` instead of computing `searchCompanions` at its call sites?